test/e2e/field_selectors/field_selectors_test.go

```diff
@@ -0,0 +1,181 @@
+		if token != "" {
+			loginArgs = append(loginArgs, "--token", token)
+		}
+		// attempt login
+		out, err := harness.CLI(loginArgs...)
+
+		// if openshift authentication is required, try to obtain a token
+		if strings.Contains(out, "You must obtain an API token by visiting") {
+			token, err = harness.SH("oc", "whoami", "-t")
+			token = strings.Trim(token, "\n")
+			Expect(err).ToNot(HaveOccurred())
+			Expect(token).ToNot(BeEmpty(), "token from oc whoami should not be empty")
+			loginArgs = append(loginArgs, "--token", token)
+			out, err = harness.CLI(loginArgs...)
+
+		}
+		Expect(err).ToNot(HaveOccurred())
+		out = strings.Trim(out, "\n.")
+		Expect(out).To(BeElementOf("Auth is disabled", "Login successful"))
+	})
+
+	AfterEach(func() {
+		harness.Cleanup(false) // do not print console on error
+	})
+
+	Context("login", func() {
+		It("should have worked, and we can list devices", func() {
+			out, err := harness.CLI("get", "devices")
+			Expect(err).ToNot(HaveOccurred())
+			Expect(out).To(ContainSubstring("NAME"))
+		})
+	})
+
+	Context("apply/recursive", func() {
+		It("should work for a complete set of yamls", func() {
+			out, err := harness.CLI("apply", "-R", "-f", filepath.Join(util.GetTopLevelDir(), "examples/device.yaml"))
+			Expect(err).ToNot(HaveOccurred())
+			// expect out to contain 200 OK or 201 Created
+			Expect(out).To(MatchRegexp(`(200 OK|201 Created)`))
+
+			// check a for a couple of the yamls we know to exist in the examples directory
+			Expect(out).To(ContainSubstring("examples/device.yaml/f62dfb4f5d2cdbb9339362b5f18f2ce268d75649bdc80003f0e04292a7ef8c3"))
+			Expect(out).To(ContainSubstring("examples/resourcesync.yaml/default-sync"))
+		})
+	})
+
+	Context("apply/fleet", func() {
+		It("create for a complete fleet", func() {
+			// make sure it doesn't exist
+			_, _ = harness.CLI("delete", "fleet/e2e-test-fleet")
+
+			out, err := harness.CLI("apply", "-f", filepath.Join(util.GetTopLevelDir(), "examples/fleet.yaml"))
+			Expect(err).NotTo(HaveOccurred())
+			Expect(out).To(ContainSubstring("201 Created"))
+		})
+	})
+
+	Context("Basic Functionality Tests", func() {
+		It("filters devices by name", func() {
+			out, err := harness.CLI("get", "devices", "--field-selector", "metadata.name=f62dfb4f5d2cdbb9339362b5f18f2ce268d75649bdc80003f0e04292a7ef8c3")
+			Expect(err).ToNot(HaveOccurred())
+			Expect(out).To(ContainSubstring("f62dfb4f5d2cdbb9339362b5f18f2ce268d75649bdc80003f0e04292a7ef8c3"))
+		})
+
+		It("filters devices by owner", func() {
+			out, err := harness.CLI("get", "devices", "--field-selector", "metadata.owner=Fleet/e2e-test-fleet")
+			Expect(err).ToNot(HaveOccurred())
+			Expect(out).To(ContainSubstring("NAME"))
+		})
+
+		It("filters devices by creation timestamp", func() {
+			out, err := harness.CLI("get", "devices", "--field-selector", "'metadata.creationTimestamp>=2024-01-01T00:00:00Z','metadata.creationTimestamp<2025-01-01T00:00:00Z'")
```

Comment:
Could you, please, use some generated timestamp, otherwise your filter will return zero in less than 1 month?